cmd/feegrant.go

```diff
@@ -0,0 +1,288 @@
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/feegrant"
+	"github.com/spf13/cobra"
+)
+
+// flag for feegrant module
+const (
+	FlagExpiration  = "expiration"
+	FlagPeriod      = "period"
+	FlagPeriodLimit = "period-limit"
+	FlagSpendLimit  = "spend-limit"
+	FlagAllowedMsgs = "allowed-messages"
+)
+
+func feegrantQueryGrantsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "grants [grantee]",
+		Short: "query things about a grantee",
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cl := config.GetDefaultClient()
+			pageReq, err := ReadPageRequest(cmd.Flags())
+
+			if err != nil {
+				return err
+			}
+
+			var address types.AccAddress
+			var keyNameOrAddress string
+
+			if len(args) == 0 {
+				keyNameOrAddress = cl.Config.Key
+			} else {
+				keyNameOrAddress = args[0]
+			}
+
+			address, err = cl.AccountFromKeyOrAddress(keyNameOrAddress)
+
+			if err != nil {
+				return err
+			}
+
+			grants, err := cl.QueryFeeGrants(address, pageReq)
+			if err != nil {
+				return err
+			}
+
+			// FIXME: if anyone can tell me why JSON output without this doesn't work gets a 5 OSMO tip
+			// FIXME: seems like there should be no side effects with using the MarshalProto function, but thats not the
+			for grant := range grants {
+				_, err := cl.MarshalProto(grants[grant])
+				if err != nil {
+					return err
+				}
+			}
+
+			return cl.PrintObject(grants)
+		},
+	}
+
+	return paginationFlags(cmd)
+}
+
+func feegrantQueryGrantCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "grant grantee [granter]",
+		Short: "query things about a single grant",
+		Args:  cobra.RangeArgs(1, 2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cl := config.GetDefaultClient()
+
+			var (
+				granterAddr      types.AccAddress
+				granteeAddr      types.AccAddress
+				err              error
+				keyNameOrAddress string
+			)
+
+			if len(args) == 1 {
+				keyNameOrAddress = cl.Config.Key
+			} else {
+				keyNameOrAddress = args[1]
+			}
+
+			granterAddr, err = cl.AccountFromKeyOrAddress(keyNameOrAddress)
+
+			if err != nil {
+				return fmt.Errorf("did not find wallet %s", keyNameOrAddress)
+			}
+
+			granteeAddr, err = cl.DecodeBech32AccAddr(args[0])
+			if err != nil {
+				return err
+			}
+
+			grant, err := cl.QueryFeeGrant(granteeAddr, granterAddr)
+
+			if err != nil {
+				return err
+			}
+
+			return cl.PrintObject(grant)
+		},
+	}
+
+	return cmd
+}
+
+func feegrantFeeGrantCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "grant grantee",
+		Short: "Grant fee allowance to an address",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				granterAddr sdk.AccAddress
+				granteeAddr sdk.AccAddress
+				err         error
+			)
+
+			cl := config.GetDefaultClient()
+			key, _ := cmd.Flags().GetString(FlagFrom)
```

Comment:
Please don't use the FROM flag, lets make these arguements in the style of the other commands.